### PR TITLE
ACM-13002-Policy-overview-with-lots-of-clusters

### PR DESCRIPTION
**Pull request by jeswanke**
@mprahl @showeimer 

This is meant to address the case where the user has lots of violating clusters in their environment:

**BEFORE:** all clusters in environment were listed alphabetically till the end--if there are 3000 clusters, it will be a very long list:
CleanShot 2024-11-14 at 12 20 07

**AFTER:** the clusters with the most violations will be sorted to the top which will naturally put the compliant clusters at the bottom which will be hidden in a collapsed section:
CleanShot 2024-11-14 at 12 23 16

If you look at Scott's dev environment,  the cluster that seems to require the most attention is towards the bottom of the list:
CleanShot 2024-11-14 at 12 28 42@2x

### CODE REVIEW

**Comment by showeimer** [comment]
Hey @jeswanke, this looks good for now; and thanks for taking the time to try to improve this.  I was aware that this card was definitely not going to scale well with high cluster counts; so I think this workaround is probably okay for now.  I am planning to have this card moved into a proper table to make this experience more first-class and simultaenously address this scale problem; so there will probably come a time where this card will be removed.  This is a decent solution for the time being, thanks again!

File: backend/src/app.ts [hunk 1]
```diff
@@ -34,6 +34,8 @@ const isProduction = process.env.NODE_ENV === 'production'
 const isDevelopment = process.env.NODE_ENV === 'development'
 const eventsEnabled = process.env.DISABLE_EVENTS !== 'true'
 
+// process.env.MOCK_CLUSTERS = '100'
```
Comment (yiraeChristineKim) [inline backend/src/app.ts:37]: Is this comment required?

File: backend/src/lib/server-side-events.ts [hunk 2]
```diff
@@ -277,12 +278,24 @@ export class ServerSideEvents {
...
+    let allEvents: ServerSideEvent[] = []
     for (const eventID in this.events) {
       if (Number(eventID) <= lastEventID) continue
-      this.sendEvent(clientID, this.events[eventID])
-      sentCount++
+      allEvents.push(this.events[eventID])
+    }
+
+    // mock a large environment
+    if (process.env.MOCK_CLUSTERS) {
```
Comment (yiraeChristineKim) [inline backend/src/lib/server-side-events.ts:288]: This is for test right?

Comment (yiraeChristineKim) [reply backend/src/lib/server-side-events.ts:288]: Retaining test code is good!

Comment (jeswanke) [reply backend/src/lib/server-side-events.ts]: I removed that--- I had added functionality to mock lots of clusters but I moved it into it's own pr

**Pull request merged**